javascript/code-challenges/tree/tree.js

```diff
@@ -9,21 +10,86 @@ class Node {
 }
 
 class BinaryTree {
-  constructor(root = null) {
-    this.root = root;
+  constructor(rootNode = null) {
+    this.rootNode = new Node(rootNode);
   }
+  preOrder() {
+    let newArray = []; // declare empty array for the output
+    let currentNode = this.rootNode;
 
+    let traverse = rootNode => {
+      newArray.push[rootNode.value];
+      if (rootNode.left) traverse(rootNode.left);
+      if (rootNode.right) traverse(rootNode.right);
+    };
+    traverse(currentNode);
+    return newArray;
+  }
+
+  inOrder() {
+    let newArray = [];
+    let currentNode = this.rootNode;
+
+    let traverse = rootNode => {
+      if (rootNode.left) traverse(rootNode.left);
+      newArray.push(rootNode.value);
+      if (rootNode.right) traverse(rootNode.right);
+    };
+    traverse(currentNode);
+    return newArray;
+  }
+
+  postOrder() {
+    let newArray = [],
+      currentNode = this.rootNode;
+
+    let traverse = rootNode => {
+      if (rootNode.left) traverse(rootNode.left);
+      if (rootNode.right) traverse(rootNode.right);
+      newArray.push(rootNode.value);
+    };
+    traverse(currentNode);
+    return newArray;
+  }
+
+}
+
+class BinarySearchTree {
+  constructor(rootNode = null) {
+    this.rootNode = new Node(rootNode);
+  }
+
+  add(newNodeValue) { // creates the add method
+    let newNode = new Node(newNodeValue);
+
+    if (!this.rootNode) {
+      this.rootNode = newNode;
+
+    } else {
+      this.add(this.rootNode, newNode);
+
+    }
+    return this.rootNode;
+  }
```

Comment:
I think that something that you need to add here, is moving to the left or right  depending on if the node is higher or lower than the current node until it gets to a point where it can be added